File: StarMines/Data/Python/green.py
def behave():
    b = Behavior()

    #check calls

    call = b.getCall()
    while call:
        if call.contents.label == b'infect':
            infect(b)
            break

        call = b.getCall()

    #movement
    pos = b.position()
    x = pos.contents.x
    y = pos.contents.y
    r = random.randint(1,4)
    if r == 1:
        x += 1.0
    if r == 2:
        x -= 1.0
    if r == 3:
        y += 1.0
    if r == 4:
        y -= 1.0

    if x > 770:
        x = 770
    if x < 0:
        x = 0
    if y > 570:
        y = 570
    if y < 0:
        y = 0

    b.setPosition(x,y)
    
    

def infect(b):
    pos = b.position()
    x = pos.contents.x
    y = pos.contents.y

    b.spawn(b'Red', x, y, None, 0)
    b.despawn(None, 0)

    b.playSound(b'infect',40,False,False)
    b.deactivate()

behave()

File: StarMines/TestData/MapGen1/Lua/Green.lua
function Green_bounds(bc)
	local pos = bc:position()
	local x = pos.x
	local y = pos.y
	if x > 770 then x = 770 end
	if x < 0 then x = 0 end
	if y > 570 then y = 570 end
	if y < 0 then y = 0 end
	bc:setPosition(x,y)
end

function Green_move(bc)
	bc:setSpeed(1)
	bc:setDir(math.random(-1,1), math.random(-1,1))
	Green_bounds(bc)
end

function Green_calls(bc)
	local call = bc:getCall()
	while call.label ~= "nil" do
		if call.label == "collision" and call.tag == "Red" then
			local pos = bc:position()
			bc:spawn("Red", pos.x, pos.y, "")
			bc:despawn("")
			bc:playSound("Infect", 40, false, false)
			return
		end
		call = bc:getCall()
	end
end

function Green(bc)
	Green_calls(bc)
	Green_move(bc)
end